src/design/messaging-api/api-types.ts

```diff
@@ -0,0 +1,663 @@
+   *   componentId: 'draggable-component',
+   *   x: 100,
+   *   y: 150
+   * });
+   * ```
+   *
+   * @see {Domain.ComponentDragStartedEvent}
+   */
+  startComponentDrag(event: Domain.ComponentDragStartedEvent): void;
+
+  /**
+   * Moves a component to a different region of a component.
+   *
+   * @param event - The component move event containing the component and region information
+   * @param event.componentId - The ID of the component to move
+   * @param event.targetComponentId - The ID of the component where the component is being moved to
+   * @param event.targetRegionId - The ID of the region that the component is being moved to
+   * @stability development
+   *
+   * @example
+   * ```typescript
+   * api.moveComponentToRegion({
+   *   componentId: 'component-123',
+   *   targetComponentId: 'parent-component',
+   *   targetRegionId: 'content-region'
+   *   sourceRegionId: 'source-content-region'
+   *   sourceComponentId: 'source-component'
+   * });
+   * ```
+   * @see {Domain.ComponentMovedToRegionEvent}
+   */
+  moveComponentToRegion(event: Domain.ComponentMovedToRegionEvent): void;
+  /**
+   * Notifies the host that a component is being hovered over.
+   *
+   * @param event - The component hover event containing the component ID
+   * @param event.componentId - The ID of the component being hovered over
+   * @stability development
+   *
+   * @example
+   * ```typescript
+   * api.hoverInToComponent({
+   *   componentId: 'component-123'
+   * });
+   * ```
+   */
+  hoverInToComponent(event: Domain.ComponentHoveredInEvent): void;
+
+  /**
+   * Notifies the host that a component is no longer being hovered over.
+   *
+   * @param event - The component hover exit event containing the component ID
+   * @param event.componentId - The ID of the component that was hovered over
+   * @stability development
+   *
+   * @example
+   * ```typescript
+   * api.hoverOutOfComponent({
+   *   componentId: 'component-123'
+   * });
+   * ```
+   */
+  hoverOutOfComponent(event: Domain.ComponentHoveredOutEvent): void;
+
+  /**
+   * Notifies the host that a component has been selected.
+   *
+   * @param event - The component selection event containing the component ID
+   * @param event.componentId - The ID of the component that was selected
+   * @stability development
+   *
+   * @example
+   * ```typescript
+   * api.selectComponent({
+   *   componentId: 'component-123'
+   * });
+   * ```
+   */
+  selectComponent(event: Domain.ComponentSelectedEvent): void;
+
+  /**
+   * Notifies the host that a component has been unselected.
+   *
+   * @param event - The component unselection event containing the component ID
+   * @param event.componentId - The ID of the component that was unselected
+   * @stability development
+   *
+   * @example
+   * ```typescript
+   * api.unselectComponent({
+   *   componentId: 'component-123'
+   * });
+   * ```
+   */
+  unselectComponent(event: Domain.ComponentUnselectedEvent): void;
```

Comment:
Is there a reason we went with unselect instead of deselect? Just curious since I don't think I've really heard unselect as a term

Reply:
I didn't even think about. Deselect makes more sense and now is the time to change it.

Reply:
renamed